Add carmine

mohammedamarnah opened the pull request.

BrunoBonacci commented:
Hi @mohammedamarnah 

it looks like it is using dynamic class loading with `Class.forName` method.

GraalVm uses the "closed world  assumption" which it means that every code path the application will run must be known at compile time.

in order to compile this you need to tell GraalVM that this image will load a class at runtime, see [Reflection](https://github.com/oracle/graal/blob/master/substratevm/Reflection.md) doc.

in order to compile this you will need to create a file called: `graalvm-config/reflect-config.json` with the list of classes dynamically loaded:

```
[
{
  "name":"org.apache.commons.pool2.impl.EvictionPolicy"
}
]
```

and pass the following config to GrralVM native image with this option:
`native-image -H:ConfigurationFileDirectories=./graalvm-config/`

this should get you passed this problem

mohammedamarnah commented:
Thanks @BrunoBonacci!
I've updated the Readme file and added building instructions. I would like to note that I had to add `"allPublicConstructors" : true` to `reflect-config` as well, because it failed again with a NoMethodFoundException for the constructor. I'm not sure if that's the best way to do it, so please let me know what you think! 😄

BrunoBonacci commented:
That's correct, thanks for the PR
